lib/jit_preloader/active_record/associations/preloader/ar7_branch.rb

```diff
@@ -0,0 +1,24 @@
+module JitPreloader
+  module PreloaderBranch
+    """
+    From ActiveRecord version >= 7.x.x, it introduces a improvement for preloading associations in batch
+    https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations/preloader.rb#L121
+
+    Previously we make sure to check and direct return if an association klass is not a valid ActiveRecord::Base
+    class through prepended methods within lib/jit_preloader/active_record/associations/preloader/arx_association.rb
+    The Batch approach move the check of association klass forward before calling individual association.
+    https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations/preloader/batch.rb#L17
+
+    This module prepend on Branch class which is the object we grab the list of loaders aka associations
+    and remove the one that does not have a valid ActiveRecord::Base klass
+    """
+
+    def loaders
+      @loaders = super.find_all do |loader|
+        loader.klass.respond_to?(:base_class)
```

Comment:
Why are we checking for the existence of the `base_class` method in order to tell if something is an ActiveRecord::Base or not? Why not do something like this instead, which is more explicit about what it's trying to accomplish:
```suggestion
        loader.klass < ::ActiveRecord::Base
```
(I may be missing something here.)

Reply:
That's technically also ok, but just to be specific to the situation, I chose to match precisely how ActiveRecord pulls the data and failed currently https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations/preloader/batch.rb#L17
But I do not have a strong opinion either way.

Reply:
Ah, I see! Hmm...

I think that, on balance, I prefer the explicit check for `ActiveRecord::Base` because it makes very clear to the reader of this code what it's trying to do. It's less clear where the `base_class` method comes from or why we expect it to be present.

Reply:
ok, that's good with me. And it does express more clearly on the intension.